fedimint-testing/src/gateway.rs

```diff
@@ -63,27 +61,26 @@ impl GatewayTest {
 
         // Create federation client builder for the gateway
         let client_builder: DynGatewayClientBuilder =
-            StandardGatewayClientBuilder::new(path.clone(), MemDbFactory.into(), address.clone())
-                .into();
+            StandardGatewayClientBuilder::new(path.clone()).into();
 
         let gatewayd_db = Database::new(MemDatabase::new(), decoders.clone());
         let gateway = Gateway::new_with_lightning_connection(
             Arc::new(lightning),
             client_builder.clone(),
-            decoders.clone(),
-            module_gens.clone(),
             DEFAULT_FEES,
             gatewayd_db,
             &mut task,
+            address.clone(),
         )
         .await
         .unwrap();
 
         gateway
             .spawn_webserver(listen, password.clone(), &mut task)
             .await;
-        task.spawn("gatewayd", move |handle| async {
-            if let Err(err) = gateway.run(handle).await {
+        let mut subgroup = task.make_subgroup().await;
+        task.spawn("gatewayd", |_| async move {
+            if let Err(err) = gateway.run(&mut subgroup).await {
```

Comment:
`run` is deprecated and going to go away since we do not need that event loop. Will remove in a subsequent PR.

Reply:
https://github.com/fedimint/fedimint/issues/2637